src/Components/reservations/NewReservation.jsx

```diff
@@ -1,37 +1,45 @@
 // Singup.js
-import React from 'react';
+import React, { useEffect } from 'react';
+import { v4 } from 'uuid';
+import { useNavigate, useParams } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
-import { fetchSignup } from '../../redux/auth/authSlice';
 import './reservation.css';
+import { addReservation } from '../../redux/reservation/reservationSlice';
+import NavigationPanel from '../NavigationPanel';
 
 export default function NewReservation() {
   const dispatch = useDispatch();
-  const { ebikes } = useSelector((state) => state.ebikeSlice);
+  const navigate = useNavigate();
+  const { id } = useParams();
+  const { isLogined } = useSelector((state) => state.authSlice);
+  const { locations } = useSelector((state) => state.locationSlice);
 
   const handleSubmit = (e) => {
     e.preventDefault();
     const f = e.currentTarget;
 
-    const arr = [f.name.value, f.email.value, f.password.value,
-      f.passwordConfirmation.value, f.role.value];
+    const arr = [f.startingDate.value, f.endingDate.value, f.city.value];
 
     if (arr.some((item) => item.trim().length === 0)) return;
 
     const data = {
-      name: arr[0],
-      email: arr[1],
-      password: arr[2],
-      password_confirmation: arr[3],
-      role: arr[4],
+      starting_date: arr[0],
+      ending_date: arr[1],
+      location: arr[2],
+      ebike_id: id,
     };
 
-    dispatch(fetchSignup(data));
+    dispatch(addReservation(data));
```

Comment:
very well done , I would just suggest using loading and and response state from the reducer to let the user know that the request is loading and redirect them to their reservations list after the reservation was successfuly created